src/nbl/ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -980,38 +968,31 @@ void CMitsubaLoader::cacheEmissionProfile(SContext& ctx, const CElementEmissionP
 {
 	if (!profile)
 		return;
-	
+
 	const auto cacheKey = ctx.emissionProfileCacheKey(*profile);
 
 	const asset::IAsset::E_TYPE types[]{ asset::IAsset::ET_IMAGE_VIEW, asset::IAsset::ET_TERMINATING_ZERO };
+
 	if (ctx.override_->findCachedAsset(cacheKey, types, ctx.inner, 0u).getContents().empty())
 	{
 		auto assetLoaded = interm_getAssetInHierarchy(m_assetMgr, profile->filename, ctx.inner.params, 0u, ctx.override_);
 
-		ICPUImageView::SCreationParams viewParams = {};
+	
 		auto contentRange = assetLoaded.getContents();
 		if (contentRange.empty())
 		{
 			os::Printer::log("[ERROR] Could Not Find Texture: " + cacheKey, ELL_ERROR);
 			return;
 		}
-		viewParams.image = core::smart_refctd_ptr_static_cast<asset::ICPUImage>(*contentRange.begin());
-		viewParams.flags = static_cast<ICPUImageView::E_CREATE_FLAGS>(0);
-		viewParams.viewType = IImageView<ICPUImage>::ET_2D;
-		viewParams.format = viewParams.image->getCreationParameters().format;
-		viewParams.subresourceRange.aspectMask = static_cast<IImage::E_ASPECT_FLAGS>(0);
-		viewParams.subresourceRange.levelCount = viewParams.image->getCreationParameters().mipLevels;
-		viewParams.subresourceRange.layerCount = 1u;
-
-		asset::SAssetBundle viewBundle(nullptr, { ICPUImageView::create(std::move(viewParams)) });
-		ctx.override_->insertAssetIntoCache(std::move(viewBundle), cacheKey, ctx.inner, 0u);
-	}
 	
-	{
-		ISampler::SParams samplerParams = ctx.emissionProfileSamplerParams(*profile);
-		const asset::IAsset::E_TYPE types[] = { asset::IAsset::ET_SAMPLER, asset::IAsset::ET_TERMINATING_ZERO };
+		auto meta = core::make_smart_refctd_ptr<asset::CIESProfileMetadata>(*static_cast<const asset::CIESProfileMetadata*>(assetLoaded.getMetadata()));
+		asset::SAssetBundle viewBundle(std::move(meta), {assetLoaded.getContents().begin()[0]});
+		ctx.override_->insertAssetIntoCache(std::move(viewBundle), cacheKey, ctx.inner, 0u);
```

Comment:
this makes no sense, you've just asked for an asset to be loaded using `interm_getAssetInHierarchy` its already inserted into the cache, the whole bundle!

Reply:
ok so you're doing all these gymnastics to "double cache" `something.ies` as `something.ies?ies`

just stop it https://github.com/Devsh-Graphics-Programming/Nabla/pull/630/files/faf4f805df55fe3af3e1a80188494c324e015919..cb61c15d7f6695c3e22f3270d9173d72f401326e#r1456676288